File: examples/python/streaming.py
import json
import os
import urllib.request


api_key = os.environ["ORIGINSTARTAI_API_KEY"]
base_url = os.environ.get("ORIGINSTARTAI_BASE_URL", "https://YOUR_7016_API_BASE_URL/v1")
model = os.environ.get("ORIGINSTARTAI_MODEL", "YOUR_DEFAULT_MODEL")

payload = {
    "model": model,
    "stream": True,
    "messages": [
        {"role": "user", "content": "Stream a short welcome message."}
    ],
}

request = urllib.request.Request(
    f"{base_url.rstrip('/')}/chat/completions",
    data=json.dumps(payload).encode("utf-8"),
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
    },
    method="POST",
)

with urllib.request.urlopen(request, timeout=60) as response:
    for line in response:
        print(line.decode("utf-8").rstrip())

File: examples/node/streaming.mjs
const apiKey = process.env.ORIGINSTARTAI_API_KEY;
const baseUrl = process.env.ORIGINSTARTAI_BASE_URL || "https://YOUR_PUBLIC_API_BASE_URL/v1";
const model = process.env.ORIGINSTARTAI_MODEL || "YOUR_ENABLED_MODEL";

if (!apiKey) {
  throw new Error("Set ORIGINSTARTAI_API_KEY first.");
}

const response = await fetch(`${baseUrl.replace(/\/$/, "")}/chat/completions`, {
  method: "POST",
  headers: {
    Authorization: `Bearer ${apiKey}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    model,
    stream: true,
    messages: [
      { role: "user", content: "Stream a short welcome message." },
    ],
  }),
});

for await (const chunk of response.body) {
  process.stdout.write(new TextDecoder().decode(chunk));
}
